### ISSUE TITLE
kASLR usage and kernel's _TEXT mapping by Checkra1n

### ISSUE THREAD

**Issue by AlexeyGorbushin** [checkra1n/BugTracker#168]
Does Checkra1n change iBoot parameters to modify kASLR behavior so that to disable kernel sliding? If no, are there plans to add such feature?

How the kernel's _TEXT segment is mapped? Do the _TEXT pages have 'write' permission to be able to apply arbitrary patches?

**Comment by Lakr233** [comment]
the patch applied to kernel should be done during the kernel load and locked after its loaded. 

it is NOT recommended to disable kpp and kASLR-randomize it is sure to lower your device's security and it is useless for normal users and if you are a developer you won't ask about this.

**Comment by AlexeyGorbushin** [comment]
> if you are a developer you won't ask about this.

Not each developer is a system or kernel level developer. That's why I am asking this questions.

**Comment by kpwn** [comment]
Right now, kASLR is unaffected in any way by checkra1n, and __TEXT is mapped normally, with both KPP and KTRR protecting it. One goal of the project is to transition to a kext loader model, where we will try to keep KPP and KTRR present, but also allow an option to get rid of them.

**Comment by AlexeyGorbushin** [comment]
@kpwn Thanks for the detailed reply!

**Comment by jorgschulze73** [comment]
It was November when we last heard about this... is there any news? I'm looking forward writing a jailbreak tweak which hooks syscalls, but unfortunately that's not possible without disabling KPP/KTRR :(

**Comment by jorgschulze73** [comment]
Pinging after another 3 months... do you guys plan to implement this ever?
Don't get me wrong, I'm really thankful for your work.
I'm just wondering if this will ever be a thing, considering the first checkra1n version happened almost 1 year ago already.

**Comment by AlexeyGorbushin** [comment]
@jorgschulze73 
Maybe the KTRW project can help you as it is able to disable KTRR permanently.
https://github.com/googleprojectzero/ktrw

**Comment by jorgschulze73** [comment]
> @jorgschulze73
> Maybe the KTRW project can help you as it is able to disable KTRR permanently.
> https://github.com/googleprojectzero/ktrw

Yes, I'm aware of KTRW but unfortunately I don't own any supported devices to use it.
Also, my idea was to work on a jb bypass tweak to release to the public... and KTRW not only works on a very limited set of devices, but also requires quite the setup for the average user to be able to use it.
